src/plugins/jsx.ts

```diff
@@ -0,0 +1,18 @@
+import type { Plugin } from 'vite'
+
+const needsJsxProcessing = (id: string = '') =>
+  !id.includes('node_modules') && ['.vue', '.jsx', '.tsx'].some(extension => id.includes(extension))
+
+export const jsxPlugin: Plugin = {
+  name: 'nuxt:jsx',
+  transform (code, id) {
+    if (needsJsxProcessing(id)) {
+      code = code.replace(/render\s*\(\s*\)\s*\{/g, 'render(h){')
```

Comment:
> Also, render does not accepts second arg?

Yes, as far as I know render only receives `h`.

I normally prefer early returning but in this case it makes it more verbose.

Reply:
Yes, of course. I think if someone needs to access the second arg, they won't need our transform because they will already likely have first arg too - e.g.:
```js
export default {
  render (h, { props }) {
    // no transform required
  }
```

Reply:
Makes sense